test/e2e/network/service.go

```diff
@@ -1971,6 +1972,88 @@ var _ = SIGDescribe("Services", func() {
 		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
 	})
 
+	ginkgo.It("should not allocate NodePorts for Type LoadBalancer if allocateLoadBalancerNodePorts=false [Feature:ServiceLBNodePortControl]", func() {
```

Comment:
this is an api only test, consider making it an integration, the other e2e will cover this.
No blocking comment :)

Reply:
good point, I'll update this to be an integration test